packages/arb-rpc-node/snapshot/snapshot.go

```diff
@@ -371,6 +371,46 @@ func (s *Snapshot) GetStorageAt(account common.Address, index *big.Int) (*big.In
 	return arbos.ParseGetStorageAtResult(res.ReturnData)
 }
 
+func (s *Snapshot) SetNonce(account common.Address, nonce uint64) error {
+	_, err := s.basicCall(arbos.SetNonceData(account, nonce), common.NewAddressFromEth(arbos.ARB_TEST_ADDRESS))
```

Comment:
I think we want the `checkValidResult` check here like in `GetBalance` to catch if the method fails for some reason. Especially with that, there's enough code here that I think a common helper method that takes just `data []byte` would be good